protocol/messenger_handler.go

```diff
@@ -1442,7 +1442,41 @@ func (m *Messenger) HandleCommunityCancelRequestToJoin(state *ReceivedMessageSta
 		return ErrInvalidCommunityID
 	}
 
-	requestToJoin, err := m.communitiesManager.HandleCommunityCancelRequestToJoin(signer, cancelRequestToJoinProto)
+	community, err := m.communitiesManager.GetByID(cancelRequestToJoinProto.CommunityId)
+
+	if err != nil {
+		return err
+	}
+
+	senderID := state.CurrentMessageState.Contact.ID
+	senderPubKey, err := common.HexToPubkey(senderID)
+
+	if err != nil {
+		return err
+	}
+
+	previousRequestToJoin, err := m.communitiesManager.GetRequestToJoinByPkAndCommunityID(senderPubKey, community.ID())
+
+	if err != nil {
+		return err
+	}
+
+	if community.HasMember(senderPubKey) && cancelRequestToJoinProto.Clock > previousRequestToJoin.Clock {
+		response, err := m.RemoveUserFromCommunity(
```

Comment:
There is still an issue with fetching community multiple times (RemoveUserFromCommunity fetches community by ID). Isn't it possible to move all this logic to `communitiesManager.HandleCommunityCancelRequestToJoin` as suggested previously?

Reply:
Done